File: Scripts/results/generated_code_results/RN_gpt/Utility&PracticalTools/file_manage/StorageDetailsScreen.js
import React from 'react';
import { View, Text, StyleSheet, Image, TouchableOpacity, ScrollView } from 'react-native';

const StorageDetailsScreen = ({ navigation }) => {
  return (
    <View style={styles.container}>
      <View style={styles.header}>
        <TouchableOpacity onPress={() => navigation.goBack()}>
          <Image source={require('../assets/snack-icon.png')} style={styles.backIcon} />
        </TouchableOpacity>
        <Text style={styles.title}>Internal Storage</Text>
        <View style={styles.placeholder} />
      </View>
      
      <ScrollView contentContainerStyle={styles.content}>
        <View style={styles.progressContainer}>
          <Image source={require('../assets/snack-icon.png')} style={styles.progressImage} />
          <Text style={styles.progressText}>66%</Text>
        </View>

        <View style={styles.storageInfo}>
          <Text style={styles.storageText}>Used</Text>
          <Text style={styles.storageText}>Available</Text>
          <Text style={styles.storageText}>Total</Text>
        </View>
        
        <View style={styles.storageInfo}>
          <Text style={styles.storageValue}>85 GB</Text>
          <Text style={styles.storageValue}>43 GB</Text>
          <Text style={styles.storageValue}>256 GB</Text>
        </View>

        <View style={styles.freeUpSpaceContainer}>
          <Image source={require('../assets/snack-icon.png')} style={styles.freeUpSpaceIcon} />
          <View style={styles.freeUpSpaceTextContainer}>
            <Text style={styles.freeUpSpaceLabel}>Free up space</Text>
            <Text style={styles.freeUpSpaceDescription}>Go to Clean to manage and free up space</Text>
          </View>
        </View>

        {renderCategory('Apps', '1,427 items 䍘 55 GB')}
        {renderCategory('Videos', '53 items 䍘 9.9 GB')}
        {renderCategory('Document', '127 items 䍘 9.0 GB')}
        {renderCategory('Images', '1,432 items 䍘 6.8 GB')}
        {renderCategory('Audio', '421 items 䍘 116 MB')}
        {renderCategory('Audio', '421 items 䍘 116 MB')}
      </ScrollView>
    </View>
  );
};

const renderCategory = (label, details) => (
  <View style={styles.categoryContainer}>
    <Image source={require('../assets/snack-icon.png')} style={styles.categoryIcon} />
    <View style={styles.categoryTextContainer}>
      <Text style={styles.categoryLabel}>{label}</Text>
      <Text style={styles.categoryDetails}>{details}</Text>
    </View>
  </View>
);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
  },
  header: {
    flexDirection: 'row',
    alignItems: 'center',
    paddingTop: 10,
    paddingHorizontal: 10,
  },
  backIcon: {
    width: 25,
    height: 25,
  },
  title: {
    flex: 1,
    textAlign: 'center',
    fontSize: 18,
    fontWeight: 'bold',
  },
  placeholder: {
    width: 25,
  },
  content: {
    paddingHorizontal: 15,
    paddingBottom: 20,
  },
  progressContainer: {
    alignItems: 'center',
    marginVertical: 20,
  },
  progressImage: {
    width: 100,
    height: 100,
  },
  progressText: {
    position: 'absolute',
    top: '40%',
    fontSize: 18,
    fontWeight: 'bold',
  },
  storageInfo: {
    flexDirection: 'row',
    justifyContent: 'space-around',
    marginVertical: 10,
  },
  storageText: {
    fontSize: 16,
  },
  storageValue: {
    fontSize: 16,
    fontWeight: 'bold',
  },
  freeUpSpaceContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    marginVertical: 15,
  },
  freeUpSpaceIcon: {
    width: 30,
    height: 30,
  },
  freeUpSpaceTextContainer: {
    marginLeft: 10,
  },
  freeUpSpaceLabel: {
    fontSize: 16,
    fontWeight: 'bold',
  },
  freeUpSpaceDescription: {
    fontSize: 14,
    color: 'grey',
  },
  categoryContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    marginVertical: 10,
  },
  categoryIcon: {
    width: 30,
    height: 30,
  },
  categoryTextContainer: {
    marginLeft: 10,
  },
  categoryLabel: {
    fontSize: 16,
    fontWeight: 'bold',
  },
  categoryDetails: {
    fontSize: 14,
    color: 'grey',
  },
});

export default StorageDetailsScreen;

File: Scripts/results/generated_code_results/Flutter/bak_code/Utility&PracticalTools/file_manage/StorageDetailsScreen.dart
import 'package:flutter/material.dart';
import 'dart:math' as math;
class StorageDetailsScreen extends StatelessWidget {
@override
Widget build(BuildContext context) {
return Scaffold(
appBar: AppBar(
leading: IconButton(
icon: Icon(Icons.arrow_back),
onPressed: () => Navigator.pop(context),
),
title: Text('Internal Storage'),
actions: [
Padding(
padding: const EdgeInsets.only(right: 16.0),
child: Text('9:41'),
),
],
),
body: SingleChildScrollView(
child: Padding(
padding: const EdgeInsets.all(16.0),
child: Column(
crossAxisAlignment: CrossAxisAlignment.start,
children: [
Center(
child: CustomPaint(
size: Size(200, 200),
painter: CircleChartPainter(0.66),
),
),
SizedBox(height: 20),
Row(
mainAxisAlignment: MainAxisAlignment.spaceAround,
children: [
_buildStorageInfo('Used', '85 GB'),
_buildStorageInfo('Available', '43 GB'),
_buildStorageInfo('Total', '256 GB'),
],
),
SizedBox(height: 20),
_buildFreeUpSpaceSection(),
SizedBox(height: 20),
_buildStorageCategory('Apps', '1,427 items', '55 GB', 0.8),
_buildStorageCategory('Videos', '53 items', '9.9 GB', 0.2),
_buildStorageCategory('Document', '127 items', '9.0 GB', 0.15),
_buildStorageCategory('Images', '1,432 items', '6.8 GB', 0.1),
_buildStorageCategory('Audio', '421 items', '116 MB', 0.05),
_buildStorageCategory('Audio', '0 items', '0 MB', 0),
],
),
),
),
);
}
Widget _buildStorageInfo(String label, String value) {
return Column(
children: [
Text(label, style: TextStyle(color: Colors.grey)),
SizedBox(height: 4),
Text(value, style: TextStyle(fontWeight: FontWeight.bold)),
],
);
}
Widget _buildFreeUpSpaceSection() {
return Container(
padding: EdgeInsets.all(16),
decoration: BoxDecoration(
color: Colors.blue[50],
borderRadius: BorderRadius.circular(8),
),
child: Row(
children: [
Icon(Icons.cleaning_services, color: Colors.blue),
SizedBox(width: 16),
Expanded(
child: Column(
crossAxisAlignment: CrossAxisAlignment.start,
children: [
Text('Free up space', style: TextStyle(fontWeight: FontWeight.bold)),
Text('Go to Clean to manage and free up space'),
],
),
),
],
),
);
}
Widget _buildStorageCategory(String title, String items, String size, double progress) {
return Padding(
padding: const EdgeInsets.symmetric(vertical: 8.0),
child: Column(
crossAxisAlignment: CrossAxisAlignment.start,
children: [
Row(
children: [
Image.asset('assets/images/template.png', width: 24, height: 24),
SizedBox(width: 8),
Text(title),
Spacer(),
Text('$items | $size'),
],
),
SizedBox(height: 4),
LinearProgressIndicator(
value: progress,
backgroundColor: Colors.grey[200],
valueColor: AlwaysStoppedAnimation<Color>(Colors.blue),
),
],
),
);
}
}
class CircleChartPainter extends CustomPainter {
final double percentage;
CircleChartPainter(this.percentage);
@override
void paint(Canvas canvas, Size size) {
final center = Offset(size.width / 2, size.height / 2);
final radius = math.min(size.width, size.height) / 2;
final bgPaint = Paint()
  ..color = Colors.grey[200]!
  ..style = PaintingStyle.stroke
  ..strokeWidth = 20;

canvas.drawCircle(center, radius, bgPaint);

final foregroundPaint = Paint()
  ..color = Colors.blue
  ..style = PaintingStyle.stroke
  ..strokeWidth = 20
  ..strokeCap = StrokeCap.round;

canvas.drawArc(
  Rect.fromCircle(center: center, radius: radius),
  -math.pi / 2,
  2 * math.pi * percentage,
  false,
  foregroundPaint,
);

final textPainter = TextPainter(
  text: TextSpan(
    text: '${(percentage * 100).toInt()}%',
    style: TextStyle(fontSize: 40, fontWeight: FontWeight.bold),
  ),
  textDirection: TextDirection.ltr,
);

textPainter.layout();
textPainter.paint(canvas, center - Offset(textPainter.width / 2, textPainter.height / 2));
}
@override
bool shouldRepaint(CustomPainter oldDelegate) => false;
}